cpp/src/copying/contiguous_split.cu

```diff
@@ -14,466 +14,1129 @@
+    output.first->is_validity          = false;
+    output.first++;
+
+    return output;
+  }
+
+  template <typename T, std::enable_if_t<std::is_same<T, cudf::list_view>::value>* = nullptr>
+  src_buf_output operator()(column_view const& col,
+                            src_buf_info* head,
+                            src_buf_output output,
+                            int parent_offset_index,
+                            int offset_depth)
+  {
+    lists_column_view lcv(col);
 
-  column_index = 0;
+    output = maybe_add_null_buffer(col, output, parent_offset_index, offset_depth);
+
+    auto offset_col = output.first;
+
+    output.first->type             = type_id::INT32;  // offsets
+    output.first->offsets          = lcv.offsets().begin<int>();
+    output.first->offset_stack_pos = output.second;
+    output.second += offset_depth;
+    output.first->parent_offsets_index = parent_offset_index;
+    output.first->is_validity          = false;
+    output.first++;
+
+    // local offsets apply to the remaining children
+    offset_depth++;
+    parent_offset_index = offset_col - head;
+    return setup_source_buf_info(
+      col.child_begin() + 1, col.child_end(), head, output, parent_offset_index, offset_depth);
+  }
+
+  template <typename T, std::enable_if_t<std::is_same<T, cudf::struct_view>::value>* = nullptr>
+  src_buf_output operator()(column_view const& col,
+                            src_buf_info* head,
+                            src_buf_output output,
+                            int parent_offset_index,
+                            int offset_depth)
+  {
+    output = maybe_add_null_buffer(col, output, parent_offset_index, offset_depth);
+    return setup_source_buf_info(
+      col.child_begin(), col.child_end(), head, output, parent_offset_index, offset_depth);
+  }
+
+  template <typename T, std::enable_if_t<std::is_same<T, cudf::dictionary32>::value>* = nullptr>
+  src_buf_output operator()(column_view const& col,
+                            src_buf_info* head,
+                            src_buf_output output,
+                            int parent_offset_index,
+                            int offset_depth)
+  {
+    CUDF_FAIL("Unsupported type");
+  }
+
+  template <typename T, std::enable_if_t<!cudf::is_compound<T>()>* = nullptr>
+  src_buf_output operator()(column_view const& col,
+                            src_buf_info* head,
+                            src_buf_output output,
+                            int parent_offset_index,
+                            int offset_depth)
+  {
+    output = maybe_add_null_buffer(col, output, parent_offset_index, offset_depth);
+
+    output.first->type             = col.type().id();
+    output.first->offsets          = nullptr;
+    output.first->offset_stack_pos = output.second;
+    output.second += offset_depth;
+    output.first->parent_offsets_index = parent_offset_index;
+    output.first->is_validity          = false;
+    output.first++;
```

Comment:
This should be a constructor call instead of setting a bunch of members individually.

Reply:
A constructor call on the src_buf_info struct (the .first field) or a constructor call on the entire src_buf_output struct?

Reply:
I think on `src_buf_info`. Something that makes clear the dependency between the input and output state.